neptune/internal/mockpoolutils.go

```diff
@@ -27,9 +33,71 @@ var ReturnZero = func(q string, bindings, rebindings map[string]string) (i int64
 	return 0, nil
 }
 
-// ReturnMalformedRequestErr is a mock implementation for NeptunePool.GetCount()
+// ReturnMalformedIntRequestErr is a mock implementation for NeptunePool.GetCount()
 // that always returns an error that is judged to be not transient by
 // neptune.isTransientError
-var ReturnMalformedRequestErr = func(q string, bindings, rebindings map[string]string) (i int64, err error) {
+var ReturnMalformedIntRequestErr = func(q string, bindings, rebindings map[string]string) (i int64, err error) {
 	return -1, errors.New(" MALFORMED REQUEST ")
 }
+
+// ReturnMalformedNilInterfaceRequestErr is a mock implementation for
+// NeptunePool functions that return  (Interface{}, error) which always returns an
+// error that is judged to be not transient by neptune.isTransientError
+var ReturnMalformedNilInterfaceRequestErr = func(q string, bindings, rebindings map[string]string) (interface{}, error) {
+	return nil, errors.New(" MALFORMED REQUEST ")
+}
+
+// ReturnThreeCodeLists is mock implementation for NeptunePool.Get() that always
+// returns a slice of three graphson.Vertex(s) of type "_code_list" and with
+// a "listID" property set to "listID0", "listID1", and "ListID2" respectively.
+var ReturnThreeCodeLists = func(query string, bindings map[string]string, rebindings map[string]string) (interface{}, error) {
+	codeLists := []graphson.Vertex{}
+	for i := 0; i < 3; i++ {
+		vertex := makeVertex("_code_list")
+		key := "listID"
+		value := fmt.Sprintf("listID_%d", i)
+		setVertexProperty(&vertex, key, value)
```

Comment:
Slight resistance at the moment - because the SetVertexProperty in `mockpoolutils` deals only with the case of the property being set being a string, and having a single (not list) value.